compiler/rustc_codegen_ssa/src/back/linker.rs

```diff
@@ -178,23 +181,53 @@ fn verbatim_args<L: Linker + ?Sized>(
     }
     l
 }
+/// Add underlying linker arguments to C compiler command, by wrapping them in
+/// `-Wl` or `-Xlinker`.
+fn convert_link_args_to_cc_args(
+    cmd: &mut Command,
+    args: impl IntoIterator<Item: AsRef<OsStr>, IntoIter: ExactSizeIterator>,
+) {
+    let args = args.into_iter();
+    if args.len() == 0 {
+        return;
+    }
```

Comment:
I kept this because the previous implementation had this check (+ the `ExactSizeIterator` bound), but I think it should be unnecessary

Reply:
It's only a perf optimization now, but it shouldn't be very useful because empty iterators shouldn't be passed here normally.

I agree that it can be removed, and the `ExactSizeIterator` bound can likely be removed too.